net-libs/libwebsockets/metadata.xml

```diff
@@ -20,14 +20,20 @@
 	<use>
 		<flag name="access-log">Support generating Apache-compatible access logs</flag>
 		<flag name="cgi">Include CGI (spawn process with network-connected stdin/out/err) APIs</flag>
+		<flag name="dir">Enable directory scanning api support</flag>
+		<flag name="file-ops">Support file operations vfs</flag>
 		<flag name="client">The client part of the library and libwebsockets-test-client</flag>
+		<flag name="extensions">Compile with extensions</flag>
 		<flag name="generic-sessions">With the Generic Sessions plugin</flag>
 		<flag name="http2">Support the HTTP/2 protocol</flag>
 		<flag name="http-proxy">Support for rewriting HTTP proxying</flag>
+		<flag name="http-basic-auth">Support Basic Auth</flag>
```

Comment:
I don't feel right about so many new USE flags, especially given a lot of them seem kind of obvious and do not add dependencies. Why not just default some of these things to on/off?

Is there a reason somebody would really want to customise their libwebsockets this much?

Reply:
Embedded systems.  So the ability to filter out as much unneeded code as possible is quite useful.  I am however wondering to what level we need to have individual knobs and not just have a single "minimal" or "embedded" flag that says "switch off everything not required for core usage".  Then rather add knobs as people request "hey, we want minimal but with xyz enabled".

Reply:
I think to enable features which don't need additional deps, users can specify them in `MYCMAKEARGS` or `CMAKE_EXTRA_CACHE_FILE` through `/etc/portage/package.env`